substrafl_examples/go_further/run_diabetes_substrafl.py

```diff
@@ -0,0 +1,561 @@
+                shared to an AggregationNode in order to compute the aggregation
+                of the different analytics.
+        """
+        df = datasamples
+        states = {
+            "n_samples": len(df),
+            "means": df.select_dtypes(include=np.number).sum().to_dict(),
+            "counts": {
+                name: series.value_counts().to_dict() for name, series in df.select_dtypes(include="category").items()
+            },
+        }
+        return states
+
+    @remote_data
+    def local_second_order_computation(self, datasamples: pd.DataFrame, shared_state: Dict):
+        """This function will use the output of the ``aggregation`` function to compute
+        locally the standard deviation of the different columns.
+
+        Args:
+            datasamples (pd.DataFrame): Pandas dataframe provided by the opener.
+            shared_state (Dict): Output of a first order analytics computation,
+                that must contain the mean.
+
+        Returns:
+            Dict: dictionary containing the local information on standard deviation
+                and number of sample. This dict will be used as a state to be shared
+                to an AggregationNode in order to compute the aggregation of the
+                different analytics.
+        """
+        df = datasamples
+        means = pd.Series(shared_state["means"])
+        states = {
+            "n_samples": len(df),
+            "std": np.power(df.select_dtypes(include=np.number) - means, 2).sum(),
+        }
+        return states
+
+    @remote
+    def aggregation(self, shared_states: List[Dict]):
+        """Aggregation function that receive a list on locally computed analytics in order to
+        aggregate them.
+        The aggregation will be a weighted average using "n_samples" as weighted coefficient.
+
+        Args:
+            shared_states (List[Dict]): list of dictionaries containing a field "n_samples",
+            and the analytics to aggregate in separated fields.
+
+        Returns:
+            Dict: dictionary containing the aggregated analytics.
+        """
+        total_len = 0
+        for state in shared_states:
+            total_len += state["n_samples"]
+
+        aggregated_values = defaultdict(lambda: defaultdict(float))
+        for state in shared_states:
+            for analytics_name, col_dict in state.items():
+                if analytics_name == "n_samples":
+                    # already aggregated in total_len
+                    continue
+                for col_name, v in col_dict.items():
+                    if isinstance(v, dict):
+                        # this column is categorical and v is a dict over
+                        # the different modalities
+                        if not aggregated_values[analytics_name][col_name]:
+                            aggregated_values[analytics_name][col_name] = defaultdict(float)
```

Comment:
Isn't this case covered by L310?

Reply:
Its going one dimension deeper than L310